imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -127,28 +149,157 @@ def couple(self) -> None:
+        self.mf6_storage = self.mf6.get_value_ptr(mf6_storage_tag)
+        self.mf6_has_sc1 = self.mf6.get_value_ptr(mf6_is_sc1_tag)[0] != 0
+        self.mf6_area = self.mf6.get_value_ptr(mf6_area_tag)
+        self.mf6_top = self.mf6.get_value_ptr(mf6_top_tag)
+        self.mf6_bot = self.mf6.get_value_ptr(mf6_bot_tag)
+        self.max_iter = self.mf6.get_value_ptr(mf6_max_iter_tag)[0]
+
+        self.msw_head = self.msw.get_value_ptr("dhgwmod")
+        self.msw_volume = self.msw.get_value_ptr("dvsim")
+        self.msw_storage = self.msw.get_value_ptr("dsc1sim")
+
+        # create mapping for metamod, should move tot mapping functions at one point maybe
+        self.svat_lookup, _ = get_svat_lookup(self.msw.working_directory)
+        table_node2svat: NDArray[np.int32] = np.loadtxt(
+            self.coupling.mf6_msw_node_map, dtype=np.int32, ndmin=2
+        )
+        node_idx = table_node2svat[:, 0] - 1
+        msw_idx = [
+            self.svat_lookup[table_node2svat[ii, 1], table_node2svat[ii, 2]]
+            for ii in range(len(table_node2svat))
+        ]
+
+        self.map_msw2mod["storage"], self.mask_msw2mod["storage"] = create_mapping(
+            msw_idx,
+            node_idx,
+            self.msw_storage.size,
+            self.mf6_storage.size,
+            Operator.SUM,
+        )
+        # MetaSWAP gives SC1*area, MODFLOW by default needs SS, convert here.
+        # When MODFLOW is configured to use SC1 explicitly via the
+        # STORAGECOEFFICIENT option in the STO package, only the multiplication
+        # by area needs to be undone
+
+        if self.mf6_has_sc1:
+            conversion_terms = 1.0 / self.mf6_area
+        else:
+            conversion_terms = 1.0 / (self.mf6_area * (self.mf6_top - self.mf6_bot))
+
+        conversion_matrix = dia_matrix(
+            (conversion_terms, [0]),
+            shape=(self.mf6_area.size, self.mf6_area.size),
+            dtype=self.mf6_area.dtype,
+        )
+        self.map_msw2mod["storage"] = conversion_matrix * self.map_msw2mod["storage"]
+
+        self.map_mod2msw["head"], self.mask_mod2msw["head"] = create_mapping(
+            node_idx,
+            msw_idx,
+            self.mf6_head.size,
+            self.msw_head.size,
+            Operator.AVERAGE,
+        )
+
+        table_rch2svat: NDArray[np.int32] = np.loadtxt(
+            self.coupling.mf6_msw_recharge_map, dtype=np.int32, ndmin=2
+        )
+        rch_idx = table_rch2svat[:, 0] - 1
+        msw_idx = [
+            self.svat_lookup[table_rch2svat[ii, 1], table_rch2svat[ii, 2]]
+            for ii in range(len(table_rch2svat))
+        ]
+
+        self.map_msw2mod["recharge"], self.mask_msw2mod["recharge"] = create_mapping(
+            msw_idx,
+            rch_idx,
+            self.msw_volume.size,
+            self.mf6_recharge.size,
+            Operator.SUM,
+        )
+
+        if self.coupling.enable_sprinkling:
```

Comment:
since we decided to skip sprinkling for now, please remove everything that has to do with sprinkling: teh pointers, the mappings, etc

Reply:
This is the sprinkling from groundwater and is part of the 'normal' coupling MSW-MF6. We skip only the sprinkling form surface water right?